packages/databricks-vscode/src/cli/BricksSyncParser.ts

```diff
@@ -103,38 +106,23 @@ export class BricksSyncParser {
                 currentLogLevel =
                     bricksLogLevelToSdk.get(typeMatch[1]) ?? currentLogLevel;
             }
-
             NamedLogger.getOrCreate(Loggers.Bricks).log(currentLogLevel, line);
+        }
+    }
 
-            this.parseForActionsInitiated(line);
-            this.parseForUploadCompleted(line);
-            this.parseForDeleteCompleted(line);
-
-            if (!this.firstSyncDone) {
-                this.parseForFirstSync(line);
-            }
-
-            // this.writeEmitter.fire writes to the pseudoterminal for the
-            // bricks sync process
-            this.writeEmitter.fire(line.trim());
-
-            // When vscode flush prints the logs from events fired here,
-            // it automatically adds a new line. Since we can reasonably expect
-            // with a high probablity that all logs in one call of this process(data) func will
-            // be flushed together, we do not add a new line at the last event
-            // to keep the new line spacing consistant
-            if (i !== logLines.length - 1) {
-                this.writeEmitter.fire("\n\r");
+    // This function processes the JSON output from bricks sync and parses it
+    // to figure out if a synchronization step is in progress or has completed.
+    public processStdout(data: string) {
+        const logLines = data.split("\n");
+        for (let i = 0; i < logLines.length; i++) {
+            const line = logLines[i];
+            try {
+                this.processLine(line);
+            } catch (error: any) {
+                // Log error?
```

Comment:
@kartikgupta-db Which logger should I use here?

Reply:
@pietern  Please use the extension logger here. 
`NamedLogger.getOrCreate(Loggers.Extension)`

Reply:
Thanks, done.